--- a/auto/types	2023-02-28 09:52:32.000000000 -0500
+++ b/auto/types	2023-03-14 09:31:46.444762882 -0400
@@ -14,6 +14,7 @@
 nxt_feature="int size"
 nxt_feature_name=NXT_INT_SIZE
 nxt_feature_run=value
+nxt_feature_run_force_result=
 nxt_feature_incs=
 nxt_feature_libs=
 nxt_feature_test="#include <stdio.h>
@@ -28,6 +29,7 @@
 nxt_feature="long size"
 nxt_feature_name=NXT_LONG_SIZE
 nxt_feature_run=value
+nxt_feature_run_force_result=
 nxt_feature_incs=
 nxt_feature_libs=
 nxt_feature_test="#include <stdio.h>
@@ -42,6 +44,7 @@
 nxt_feature="long long size"
 nxt_feature_name=NXT_LONG_LONG_SIZE
 nxt_feature_run=value
+nxt_feature_run_force_result=
 nxt_feature_incs=
 nxt_feature_libs=
 nxt_feature_test="#include <stdio.h>
@@ -56,6 +59,7 @@
 nxt_feature="void * size"
 nxt_feature_name=NXT_PTR_SIZE
 nxt_feature_run=value
+nxt_feature_run_force_result=
 nxt_feature_incs=
 nxt_feature_libs=
 nxt_feature_test="#include <stdio.h>
@@ -76,6 +80,7 @@
 nxt_feature="size_t size"
 nxt_feature_name=NXT_SIZE_T_SIZE
 nxt_feature_run=value
+nxt_feature_run_force_result=
 nxt_feature_incs=
 nxt_feature_libs=
 nxt_feature_test="#include <stdio.h>
@@ -90,6 +95,7 @@
 nxt_feature="off_t size"
 nxt_feature_name=NXT_OFF_T_SIZE
 nxt_feature_run=value
+nxt_feature_run_force_result=
 nxt_feature_incs=
 nxt_feature_libs=
 nxt_feature_test="#define _FILE_OFFSET_BITS  64
@@ -106,6 +112,7 @@
 nxt_feature="time_t size"
 nxt_feature_name=NXT_TIME_T_SIZE
 nxt_feature_run=value
+nxt_feature_run_force_result=
 nxt_feature_incs=
 nxt_feature_libs=
 nxt_feature_test="#include <time.h>
