--- a/auto/modules/python
+++ b/auto/modules/python
@@ -50,6 +50,7 @@ fi
 
 . $NXT_AUTOCONF_DATA
 
+echo=echo
 
 NXT_PYTHON_CONFIG=${NXT_PYTHON_CONFIG=python-config}
 NXT_PYTHON=${NXT_PYTHON_CONFIG%-config*}
@@ -62,14 +63,18 @@ $echo "configuring Python module ..." >> $NXT_AUTOCONF_ERR
 
 nxt_found=no
 
-if /bin/sh -c "$NXT_PYTHON_CONFIG --prefix" >> $NXT_AUTOCONF_ERR 2>&1; then
+if /bin/sh -c "pkg-config python3 --exist" >> $NXT_AUTOCONF_ERR 2>&1; then
 
-    if ${NXT_PYTHON_CONFIG} --embed >/dev/null 2>&1; then
-        NXT_PYTHON_CONFIG="${NXT_PYTHON_CONFIG} --embed"
+    if pkg-config python3-embed --exist >/dev/null 2>&1; then
+        NXT_PYTHON_CONFIG="pkg-config python3-embed"
+    else
+        NXT_PYTHON_CONFIG="pkg-config python3"
     fi
 
-    NXT_PYTHON_INCLUDE=`${NXT_PYTHON_CONFIG} --includes`
-    NXT_PYTHON_LIBS=`${NXT_PYTHON_CONFIG} --ldflags`
+    NXT_PYTHON_INCLUDE=`${NXT_PYTHON_CONFIG} --cflags`
+    NXT_PYTHON_LIBS=`${NXT_PYTHON_CONFIG} --libs`
+    NXT_PYTHON_VERSION=`${NXT_PYTHON_CONFIG} --modversion`
+    NXT_PYTHON_MODULE=python${NXT_PYTHON_VERSION}
 
     if [ "$NXT_PYTHON_LIB_PATH" != "" ]; then
         # "python-config --ldflags" may not contain path to libpython.
@@ -108,6 +113,7 @@ fi
 nxt_feature="Python version"
 nxt_feature_name=""
 nxt_feature_run=value
+nxt_feature_run_force_result="${NXT_PYTHON_VERSION}"
 nxt_feature_incs="${NXT_PYTHON_INCLUDE}"
 nxt_feature_libs="${NXT_PYTHON_LIBS} $NXT_PYTHON_LDFLAGS"
 nxt_feature_test="
@@ -132,33 +138,11 @@ fi
 
 NXT_PYTHON_MOUNTS_HEADER=nxt_${NXT_PYTHON_MODULE}_mounts.h
 
-$NXT_PYTHON -c 'import os.path
-import sys
-pyver = "python" + str(sys.version_info[0]) + "." + str(sys.version_info[1])
-
-print("static const nxt_fs_mount_t  nxt_python_mounts[] = {")
-
-pattern = "{(u_char *) \"%s\", (u_char *) \"%s\", NXT_FS_BIND, (u_char *) \"bind\", 0, NULL, 1, 1},"
-base = None
-for p in sys.path:
-    if len(p) > 0:
-        if os.path.basename(p) == pyver:
-            base = p
-
-if base is None:
-    raise Exception("failed to compute sys.path mount points")
-
-print(pattern % (base, base))
-
-for p in sys.path:
-    if len(p) > 0:
-        if not p.startswith(base):
-            print(pattern % (p, p))
-
-print("};\n\n")
-
-' > $NXT_BUILD_DIR/$NXT_PYTHON_MOUNTS_HEADER
-
+cat << END > $NXT_BUILD_DIR/$NXT_PYTHON_MOUNTS_HEADER
+static const nxt_fs_mount_t  nxt_python_mounts[] = {
+{(u_char *) "/usr/lib/python${NXT_PYTHON_VERSION}", (u_char *) "/usr/lib/python${NXT_PYTHON_VERSION}", NXT_FS_BIND, (u_char *) "bind", 0, NULL, 1, 1},
+};
+END
 
 $echo " + Python module: ${NXT_PYTHON_MODULE}.unit.so"
 
