--- a/auto/ssltls	2023-03-14 21:31:12.806187446 -0400
+++ b/auto/ssltls	2023-03-14 21:31:07.878123343 -0400
@@ -37,6 +37,7 @@
         nxt_feature="OpenSSL version"
         nxt_feature_name=NXT_HAVE_OPENSSL_VERSION
         nxt_feature_run=value
+        nxt_feature_run_force_result="`pkg-config openssl --modversion`"
         nxt_feature_test="#include <openssl/ssl.h>
 
                           int main(void) {
