--- a/auto/pcre	2023-04-21 09:25:51.000000000 
+++ b/auto/pcre	2023-04-21 09:22:54.000000000 
@@ -7,10 +7,10 @@
 NXT_HAVE_PCRE2=NO
 
 if [ $NXT_TRY_PCRE2 = YES ]; then
-    if /bin/sh -c "(pcre2-config --version)" >> $NXT_AUTOCONF_ERR 2>&1; then
+    if /bin/sh -c "(pkg-config libpcre2-8 --exists)" >> $NXT_AUTOCONF_ERR 2>&1; then
 
-        NXT_PCRE_CFLAGS=`pcre2-config --cflags`
-        NXT_PCRE_LIB=`pcre2-config --libs8`
+        NXT_PCRE_CFLAGS=`pkg-config libpcre2-8 --cflags`
+        NXT_PCRE_LIB=`pkg-config libpcre2-8 --libs`
 
         nxt_feature="PCRE2 library"
         nxt_feature_name=NXT_HAVE_PCRE2
@@ -32,16 +32,16 @@
 
         if [ $nxt_found = yes ]; then
             NXT_HAVE_PCRE2=YES
-            $echo " + PCRE2 version: `pcre2-config --version`"
+            $echo " + PCRE2 version: `pkg-config libpcre2-8 --modversion`"
         fi
     fi
 fi
 
 if [ $nxt_found = no ]; then
-    if /bin/sh -c "(pcre-config --version)" >> $NXT_AUTOCONF_ERR 2>&1; then
+    if /bin/sh -c "(pkg-config libpcre --exist)" >> $NXT_AUTOCONF_ERR 2>&1; then
 
-        NXT_PCRE_CFLAGS=`pcre-config --cflags`
-        NXT_PCRE_LIB=`pcre-config --libs`
+        NXT_PCRE_CFLAGS=`pkg-config libpcre --cflags`
+        NXT_PCRE_LIB=`pkg-config libpcre --libs`
 
         nxt_feature="PCRE library"
         nxt_feature_name=NXT_HAVE_PCRE
@@ -61,7 +61,7 @@
         . auto/feature
 
         if [ $nxt_found = yes ]; then
-            $echo " + PCRE version: `pcre-config --version`"
+            $echo " + PCRE version: `pkg-config libpcre --modversion`"
         fi
     fi
 fi
