--- a/auto/options	2023-04-26 09:24:39.930925893 -0400
+++ b/auto/options	2023-04-26 09:19:17.950967249 -0400
@@ -46,6 +46,8 @@
 
 NXT_FORCE_ENDIANNESS=
 
+SUMMIT_OPEN_CERTS_RDONLY=NO
+
 for nxt_option
 do
     case "$nxt_option" in
@@ -107,6 +109,8 @@
  
         --force-endianness=*)            NXT_FORCE_ENDIANNESS="$value" ;;
 
+        --open-certs-readonly)           SUMMIT_OPEN_CERTS_RDONLY=YES        ;;
+
         --help)
             . auto/help
             exit 0
--- a/configure	2023-04-26 09:24:39.870925899 -0400
+++ b/configure	2023-04-26 09:19:02.100967654 -0400
@@ -100,6 +100,10 @@
     nxt_have=NXT_HAVE_UNIX_DOMAIN . auto/have
 fi
 
+if [ $SUMMIT_OPEN_CERTS_RDONLY = YES ]; then
+    nxt_have=SUMMIT_OPEN_CERTS_RDONLY . auto/have
+fi
+
 NXT_LIBRT=
 
 . auto/endian
--- a/src/nxt_cert.c	2023-02-28 09:52:32.000000000 -0500
+++ b/src/nxt_cert.c	2023-04-26 09:14:42.120980339 -0400
@@ -1172,8 +1172,13 @@
     p = nxt_cpymem(file.name, rt->certs.start, rt->certs.length);
     p = nxt_cpymem(p, name.start, name.length + 1);
 
+#if (SUMMIT_OPEN_CERTS_RDONLY)
+    ret = nxt_file_open(task, &file, NXT_FILE_RDONLY, NXT_FILE_OPEN,
+                        NXT_FILE_OWNER_ACCESS);
+#else
     ret = nxt_file_open(task, &file, NXT_FILE_RDWR, NXT_FILE_CREATE_OR_OPEN,
                         NXT_FILE_OWNER_ACCESS);
+#endif
 
     nxt_free(file.name);
 
